packages/@aws-cdk/aws-efs/lib/access-point.ts

```diff
@@ -109,23 +114,62 @@ export interface AccessPointProps extends AccessPointOptions {
   readonly fileSystem: IFileSystem;
 }
 
+/**
+ * Attributes that can be specified when importing an AccessPoint
+ */
+export interface AccessPointAttributes {
+  /**
+   * The ID of the AccessPoint
+   * One of this, of {@link accessPointArn} is required
+   *
+   * @default - no access point id
+   */
+  readonly accessPointId?: string;
+
+  /**
+   * The ARN of the AccessPoint
+   * One of this, of {@link accessPointId} is required
+   *
+   * @default - no access point ARN
+   */
+  readonly accessPointArn?: string;
+
+  /**
+   * The EFS filesystem
+   *
+   * @default - no EFS filesystem
+   */
+  readonly fileSystem?: IFileSystem;
+}
+
+abstract class AccessPointBase extends Resource implements IAccessPoint {
+  /**
+   * The ARN of the Access Point
+   * @attribute
+   */
+  public abstract readonly accessPointArn: string;
+
+  /**
+   * The ID of the Access Point
+   * @attribute
+   */
+  public abstract readonly accessPointId: string;
+
+  /**
+   * The filesystem of the access point
+   */
+  public abstract readonly fileSystem?: IFileSystem;
+}
+
 /**
  * Represents the AccessPoint
  */
-export class AccessPoint extends Resource implements IAccessPoint {
+export class AccessPoint extends AccessPointBase {
   /**
    * Import an existing Access Point
    */
-  public static fromAccessPointId(scope: Construct, id: string, accessPointId: string): IAccessPoint {
```

Comment:
This API should not be removed. Keep the `fromAccessPointId()` alongside `fromAccessPointAttributes()`.